### PR TITLE
chore:  Add linting for TypeScript

### PR DESCRIPTION
**Pull request by alextranwork**
### 🧪 QA
Use YALC with web-app and run `eslint --ext .js,.ts,.tsx src`, see https://github.com/heydoctor/web-app/pull/421/files . May have to temporarily install some node modules to test because of how YALC works.

### 📈 Risk Analysis
✅ Does this affect patients?
✅ Does this affect clinical?
✅ Does this mutate data?
✅ Does it touch PHI?
✅ Does it touch payments?

### CODE REVIEW

File: react-ts.json [hunk 1]
```diff
@@ -0,0 +1,76 @@
+{
```
Comment (alextranwork) [inline react-ts.json:1]: Almost a duplicate of react.json

Comment (alextranwork) [reply react-ts.json:1]: Created a new file to prevent any breaking changes.

File: react-ts.json [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    "react/jsx-wrap-multilines": 0,
+    "react/jsx-indent": 0,
+    "jsx-a11y/anchor-is-valid": 0,
+    "jsx-a11y/click-events-have-key-events": 0,
+    "jsx-a11y/no-static-element-interactions": 0,
+    "jsx-a11y/label-has-for": 0,
+    "jsx-a11y/label-has-associated-control": 0,
+    "babel/new-cap": 1,
+    "jest/no-identical-title": "error",
+    "prettier/prettier": [
```
Comment (alextranwork) [inline react-ts.json:63]: I think we should separate prettier from eslint so we can have a .prettierrc so that editors/others can use it.

Comment (choochootrain) [reply react-ts.json:63]: all major editors have linting plugins for `eslint` which provides a superset of prettier's linting functionality, is there a specific reason to have editors interface with prettier directly?

Comment (alextranwork) [reply react-ts.json:63]: I can't think of any specific reason.  We can continue using eslint to handle code styling.

**Review by choochootrain — APPROVED** [review]

**Pull request merged**